src/libcore/iter.rs

```diff
@@ -8,51 +8,278 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! Composable external iterators
+//! Composable external iteratoration
 //!
-//! # The `Iterator` trait
+//! If you've found yourself with a collection of some kind, and needed to perform an operation on
```

Comment:
Ah yes! You're right. I think one selling feature of iterators is that you can use them on things of cyclic or infinite nature.